src/Public/js/nipsweb.librarypane.js

```diff
@@ -143,36 +168,54 @@ $(document).ready(function () {
               $("#baps-channel-res").append(
                 "<li id='ManagedDB-" + data[file].managedid +
                 "' length='" + data[file].length +
-                "' title='" + data[file].title + "(" + data[file].length + ")" +
+                "' title='" + escapeHTML(data[file].title) + " (" + data[file].length + ")" +
                 "' channel='res' weight='0' type='aux' managedid='" + data[file].managedid + "'>" +
-                data[file].title + "</li>"
+                escapeHTML(data[file].title) + "</li>"
               );
             }
           }
-          $("#res-loading").hide();
           //Enable name filtering
           ulsort.List.Filter("#res-filter-name", "#baps-channel-res>li");
           //Make them activatable
           planner.registerItemClicks();
+          showAlert("Loading Library Complete", "success");
         }
       });
-      $("#res-filter-artist, #res-filter-track").hide();
+      $("#res-filter-artist-container, #res-filter-track").fadeOut();
       $("#res-filter-name").fadeIn();
     }
     //Clear the current list
     $("#baps-channel-res").empty();
     //Makes the artist search autocompleting. When an artist is selected it'll filter
     var artistLookup = new Bloodhound({
-      datumTokenizer: Bloodhound.tokenizers.obj.whitespace("title"),
+      datumTokenizer: function (datum) {
+        return Bloodhound.tokenizers.whitespace(datum.title);
+      },
       queryTokenizer: Bloodhound.tokenizers.whitespace,
       limit: 5,
       dupDetector: function (remote, local) {
         return local.title == remote.title;
       },
       prefetch: {
```

Comment:
We've previously discussed that prefetch could be the cause of some of our issues with search. Maybe pull that block out?